Dockerfile

```
@@ -0,0 +1,45 @@
+# syntax=docker/dockerfile:1
+FROM python:3.11-slim
+
+# Set build-time variables for UID and GID with defaults
+ARG UID=1000
+ARG GID=1000
+ARG USERNAME=appuser
+
+# Install system dependencies as root
+RUN apt-get update && apt-get install -y --no-install-recommends \
```

Hadolint

[warning] Pin versions in apt get install. Instead of `apt-get install <package>` use `apt-get install <package>=<version>`

(DL3008)